### PR TITLE
Request user to add bot to their server when importing roles

### PR DESCRIPTION
**Pull request by mattcasey**

### CODE REVIEW

File: charmClient.ts [hunk 1]
```diff
@@ -162,7 +162,7 @@ class CharmClient {
   }
 
   listDiscordServers (payload: { code: string }) {
-    return http.GET<{servers: DiscordUserServer[]}>('/api/discord/listServers', payload);
+    return http.GET<DiscordUserServer[]>('/api/discord/listServers', payload);
```
Comment (mattcasey) [inline charmClient.ts:165]: just simplifying.. all our endpoints return a top-level list

File: components/settings/roles/components/ImportDiscord/ImportDiscordRolesButton.tsx [hunk 2]
```diff
@@ -19,10 +20,11 @@ export default function ImportDiscordRolesButton ({ onUpdate }: { onUpdate: () =
   const [user] = useUser();
   const router = useRouter();
   const [isModalOpen, setIsModalOpen] = useState(false);
-  // See lib/discord/handleDiscordResponse.ts for the case when we return the error that will be used here
-  const [discordErrorFix, setDiscordErrorFix] = useState<{error: string, redirectLink: string} | null>(null);
-  const [discordServers, setDiscordServers] = useState<DiscordUserServer[]>([]);
   const shouldRequestServers = currentSpace && typeof router.query.code === 'string' && router.query.discord === '1' && router.query.type === 'server';
+  const { data: discordServers, error } = useSWRImmutable(
```
Comment (mattcasey) [inline components/settings/roles/components/ImportDiscord/ImportDiscordRolesButton.tsx:24]: experimenting with swr

File: pages/api/discord/importRoles.ts [hunk 3]
```diff
@@ -59,16 +59,6 @@ async function importRoles (req: NextApiRequest, res: NextApiResponse<ImportRole
     return;
   }
 
-  const userId = req.session.user.id;
-  const user = await prisma.user.findUnique({
```
Comment (mattcasey) [inline pages/api/discord/importRoles.ts:63]: dead code

File: pages/api/discord/login.ts [hunk 4]
```diff
@@ -24,7 +24,7 @@ async function login (req: NextApiRequest, res: NextApiResponse) {
     type: query.type
   }));
 
-  const oauthUrl = `${discordUrl}&${query.type === 'connect' ? '&scope=identify' : '&scope=guilds'}&state=${state}&redirect_uri=${encodeURIComponent(req.headers.host!.startsWith('localhost') ? `http://${req.headers.host}/api/discord/callback` : 'https://app.charmverse.io/api/discord/callback')}`;
+  const oauthUrl = `${discordUrl}${query.type === 'connect' ? 'prompt=consent&scope=identify' : '&permissions=0&scope=guilds%20bot'}&state=${state}&redirect_uri=${encodeURIComponent(req.headers.host!.startsWith('localhost') ? `http://${req.headers.host}/api/discord/callback` : 'https://app.charmverse.io/api/discord/callback')}`;
```
Comment (mattcasey) [inline pages/api/discord/login.ts:27]: the fix! I had to remove 'prompt=consent' or i'd get an error

**Review by Devorein — APPROVED** [review]

**Review by Devorein — APPROVED** [review]

**Pull request merged**